sqlgen/db_generator.go

```diff
@@ -26,6 +26,23 @@ func (s *State) GenNewCTE() *CTE {
 	}
 }
 
+func (s *State) Clone() *State {
+	s1 := *s
+	s1.tables = make([]*Table, 0, len(s.tables))
+	for _, tbl := range s.tables {
+		s1.tables = append(s1.tables, tbl.Clone(nil,nil,nil))
+	}
+	s1.scope = make([]map[ScopeKeyType]ScopeObj, 0, len(s.scope))
+	for _, v := range s.scope {
+		temp := map[ScopeKeyType]ScopeObj{}
+		for k, v := range v {
+			temp[k] = v
+		}
+		s1.scope = append(s1.scope, temp)
+	}
```

Comment:
This seems incorrect because the scope object can be a pointer that pointing to any entity(`*Column`, `*Table` ...) in the old state.

Reply:
do you have any ideas?

Reply:
We can return a `nil` pointer if the `len(s.scope) != 1`(it is in the middle state of generation) to indicate `Clone()` failed.